### PR TITLE
fix: avoid sending empty password property in requests for database user resource

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Proposed changes

_Jira ticket:_ [INTMDB-1230](https://jira.mongodb.org/browse/INTMDB-1230) and [HELP-51534](https://jira.mongodb.org/browse/HELP-51534)

This change ensures that `password` field is not included in create or update requests if it is empty. Includes new contract tests for AWS IAM types.

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [x] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [x] Published to AWS private registry
- [x] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [x] Validated in Atlas UI
- [x] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

## Further comments

<details>
  <summary>Manual testing stack using private registry</summary>

Stack creation using resource template:
Screenshot 2023-10-26 at 15 33 21

Verifying Atlas UI
create database users

</details>

### CODE REVIEW

File: cfn-resources/database-user/cmd/resource/resource.go [hunk 1]
```diff
@@ -48,14 +47,12 @@ func validateModel(fields []string, model *Model) *handler.ProgressEvent {
 // Create handles the Create event from the Cloudformation service.
 func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	setup()
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
```
Comment (colm-quinn) [inline cfn-resources/database-user/cmd/resource/resource.go:50]: Nice flyby

File: cfn-resources/database-user/cmd/resource/resource.go [hunk 2]
```diff
@@ -376,7 +367,7 @@ func setModel(currentModel *Model) (*admin.CloudDatabaseUser, error) {
 		DeleteAfterDate: util.StringPtrToTimePtr(currentModel.DeleteAfterDate),
 	}
 
-	if currentModel.Password != nil {
+	if util.IsStringPresent(currentModel.Password) {
```
Comment (colm-quinn) [inline cfn-resources/database-user/cmd/resource/resource.go:370]: 🎈

**Review by colm-quinn — APPROVED** [review]
LGTM, in particular great to see a test to prevent regressions.

**Comment by colm-quinn** [comment]
@AgustinBettati Once merged, let's publish to all regions. cc @marcosuma @Zuhairahmed for awareness.

**Comment by lantoli** [comment]
> LGTM, in particular great to see a test to prevent regressions.

+100 about adding regression tests

**Review by lantoli — APPROVED** [review]

**Pull request merged**